specification/compute/resource-manager/Microsoft.Compute/preview/2020-09-30/gallery.json

```diff
@@ -0,0 +1,2836 @@
+      "description": "Specifies information about the Shared Image Gallery that you want to update."
+    },
+    "GalleryProperties": {
+      "properties": {
+        "description": {
+          "type": "string",
+          "description": "The description of this Shared Image Gallery resource. This property is updatable."
+        },
+        "identifier": {
+          "$ref": "#/definitions/GalleryIdentifier"
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "title": "The current state of the gallery.",
+          "description": "The provisioning state, which only appears in the response.",
+          "enum": [
+            "Creating",
+            "Updating",
+            "Failed",
+            "Succeeded",
+            "Deleting",
+            "Migrating"
+          ]
+        },
+        "sharingProfile": {
+          "$ref": "#/definitions/SharingProfile"
+        }
+      },
+      "description": "Describes the properties of a Shared Image Gallery."
+    },
+    "GalleryIdentifier": {
+      "properties": {
+        "uniqueName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique name of the Shared Image Gallery. This name is generated automatically by Azure."
+        }
+      },
+      "description": "Describes the gallery unique name."
+    },
+    "SharingProfile": {
+      "description": "Profile for gallery sharing to subscription or tenant",
+      "properties": {
+        "permissions": {
+          "type": "string",
+          "description": "This property allows you to specify the permission of sharing gallery. <br><br> Possible values are: <br><br> **Private** <br><br> **Groups**",
+          "enum": [
+            "Private",
+            "Groups"
+          ],
+          "x-ms-enum": {
+            "name": "GallerySharingPermissionTypes",
+            "modelAsString": false
+          }
+        },
+        "groups": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharingProfileGroup"
+          },
+          "description": "A list of sharing profile groups."
+        }
+      }
+    },
+    "SharingProfileGroup": {
+      "description": "Group of the gallery sharing profile",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "This property allows you to specify the type of sharing group. <br><br> Possible values are: <br><br> **Subscriptions** <br><br> **AADTenants**",
+          "enum": [
+            "Subscriptions",
+            "AADTenants"
+          ],
+          "x-ms-enum": {
+            "name": "SharingProfileGroupTypes",
+            "modelAsString": false
+          }
+        },
+        "ids": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of subscription/tenant ids the gallery is aimed to be shared to."
```

Comment:
What permissions do you need on the target subscriptions/tenants to share a gallery with them?

Reply:
Currently we have only two permissions, private and group. Private means gallery is private and unsharable. And group permission is the one needed for sharing galleries.

Reply:
I am talking about RBAC.  What prevents me from choosing a random subscription that I don't have access to and sharing a gallery to it?

Reply:
So our feature is called as 'direct sharing' and it does not use RBAC at all. Nothing would prevent you from choosing a random subscription and share a gallery to it. It would be decided by the gallery owner entirely what subs to share to.

Reply:
Thats not appropriate.  There should be no way for users to cause something to appear in some other user's subscriptions that they have no access to.  They could put all sorts of bad stuff in descriptions, etc...  or attempt to pollute the sharedgalleries in a subscription/tenant with a bunch of garbage.  Either the user sharing the gallery needs a certain permission on the destination or they are sharing globally and there is a vetting process before your team allows a gallery to become a shared gallery

Reply:
First, we do not have create anything in other user's subscription. Let's say, I own this gallery A in my subscription S1, you are under subscription S2 and want to use it. I add your subscription info to my gallery A. So the change happens in S1 consistently. When you want to use my gallery A, you call my API and find 'wow, i have access'. You can get things you want and do deployment. **So only I, the gallery owner, has the option to modify the sharing profile of gallery A**(share to S2 or unshare to S2). If you want access to gallery A, you need to talk to me. Even I share the gallery to you and do not tell you that, you won't know unless you call my API and find you have access. In the future, we want to share some gallery globally and that would be painful if there is such permission check.

Reply:
What is "my API" in your scenario explicitly?  When the user in S2 wants to see the shared galleries they have access to what is the subscription they use in the URI?

If its S2 then yes, the user in S1 is making something show up in S2 without any access to S2 which is not OK.  If its S1 how (specifically) are you doing authz to determine if the caller should be able to see the shared gallery in S1?

Reply:
'My Api' means these listing shared gallery apis and get shared gallery apis. User in S2 uses S2 to fetch shared gallery. But I think S2 only can read the shared gallery and can deploy using shared gallery. There is no way this subscription can change anything of the shared gallery. So there is no pollution threat or any other concern about that. We sanitize response strictly.

Reply:
I started an email thread about this pr, thanks!

Reply:
> I am talking about RBAC. What prevents me from choosing a random subscription that I don't have access to and sharing a gallery to it?

Just like RBAC doesnt prevent me sharing to a random user, this wont prevent me sharing to a random subscription. Sharing is an explicit act, just like it is in RBAC.